src/abstractdataframe/iteration.jl

```diff
@@ -454,7 +454,17 @@ function mapcols!(f::Union{Function, Type}, df::DataFrame)
     if len_min != len_max
         throw(DimensionMismatch("lengths of returned vectors must be identical"))
     end
-    _columns(df) .= vs
+
+    for (i, col) in enumerate(vs)
+        fic = firstindex(col)
+        fic != 1 && _onebased_check_error(i, fic)
```

Comment:
Given that this pattern happens in lots of places and that we don't care about performance when an error happens, I would suggest changing `_onebased_check_error` to take `col`:
```suggestion
        firstindex(col) != 1 && _onebased_check_error(i, col)
```

Reply:
But `col` is a vector. What would you then print in the error message?